framework/src/media/MediaRecorderImpl.cpp

```diff
@@ -187,7 +187,7 @@ void MediaRecorderImpl::prepareRecorder(recorder_result_t& ret)
 		return notifySync();
 	}
 
-	if (mDuration > 0) {
+	if (mDuration >= 0) {
```

Comment:
Instead of this, we had better return error if duration is lower than 1.

Reply:
@Taejun-Kwon But, zero means endless recording. I think that error return should be reconsidered.
And when I think about agian, actually, eqaul condition is meaningless.
Because After calling `prepare()`, `setDuration()` is useless.

Reply:
I mean if setDuration(0) comes, return error.

Reply:
@Taejun-Kwon 
First, Returning error when setDuration(0) come must be reconsidered.

On current version, As I mentioned that zero means endless(or infinite) recording.

Let's think about below case.

> setDuration(3);
> setDuration(0);
> prepare();
> start();

Above mediarecorder should record for 3 seconds.
Because of the current comparison condition.
To change from 3 seconds to 0 seconds on current version, you need to do :

> setDuration(3);
> prepare();
> unprepare();
> prepare();
> start();

This case does not occur in static situation(or code), but in dynamic.
For supporting to set 0 sec(actually, infinite recording) without prepare()&unprepare(), we should change the comparison condition of setRecorderDuration.

Second, This change is meaningless. 
Because this line is the prepare() code. If mDuration is 0, mTotalFrame is also 0. 
This line will be meaningfuly only if mDuration is greater than 0.

Reply:
I think there was miscommunication between you and I.
So you want to set 0 as a deactivates limitation, right?
what I meaning is that 
Not 'if (mDuration ==0)', I mean 'if (second == 0)'
In your above case, of course it should play for 3 secs but when it calls setduration(0), it will return error regarding invalid param directly.
But I think disable limit values also good options.

Reply:
@Taejun-Kwon 
I made a new PR that contains new APIs and contents of this review.